### ISSUE TITLE
Urls are played in wrong  sequence exoplayer 2.9.1

### ISSUE THREAD

**Issue by CreeschYT** [google/ExoPlayer#5957]
###  Content Description
Hi, umm it's me again...
soo, I'm currently developing a LiveStream m3u player which uses exoplayer 2.9.1. When the App starts you need to click on the 'open' button which gives a MaterialFilePicker (Openfiledialog).

Now you need to choose a m3u file. All of this works. Because exoplayer can't play m3u files I used a BufferedReader to read all the lines. To actually play the channels I have to put every line that startswith "h"(url) in a List.

In this case an Arraylist. I made a Mediasource for every url with a for loop and put them into a concatenatingMediaSource. Now the urls are playing. If I click on the next button the same channel plays. Then I clicked the next button again. Now it plays channel 2. Then Channel 1, then channel 2 and then channel 3. And so on: Channel1, Channel 2, Channel 3, Channel 4 and goes back to Channel 1 ... The previous Button has to be clicked twice to go back. 

And after some time (30-40 secs) the next video plays.

This is the Code:

```
 ArrayList<String> lineStore = new ArrayList<String>();
    ConcatenatingMediaSource concatenatingMediaSource;

  protected void onActivityResult(int requestCode, int resultCode, Intent data) {


        if (requestCode == 1000 && resultCode == RESULT_OK) {
            String filePath = data.getStringExtra(FilePickerActivity.RESULT_FILE_PATH);
            // Do anything with file

            DefaultHttpDataSourceFactory dataSourceFactory = new DefaultHttpDataSourceFactory("exoplayer_video");


            exoPlayerView.setVisibility(View.VISIBLE);
            btn.setVisibility(View.INVISIBLE);
            txt.setVisibility(View.INVISIBLE);
            list.setVisibility(View.VISIBLE);

            List<ExtractorMediaSource> mediaSources = new ArrayList<>();

          //  ArrayList<String> lineStore = new ArrayList<String>();

            try {
                BufferedReader br = new BufferedReader(new FileReader(filePath));
                String line;

                while ((line = br.readLine()) != null) {
                   if (line.startsWith("h")){
                       lineStore.add(line);



                       for (String url : lineStore) {
                           mediaSources.add(new ExtractorMediaSource.Factory(dataSourceFactory).createMediaSource(Uri.parse(url)));

                       }


                   }

                }
                br.close();
            } catch (Exception e) {
                Log.e("MainAcvtivity"," exoplayer error "+ e.toString());
            }

            concatenatingMediaSource = new ConcatenatingMediaSource(mediaSources.toArray(new MediaSource[mediaSources.size()]));
            exoPlayer.prepare(concatenatingMediaSource);
            exoPlayer.setPlayWhenReady(true);


        }
    }

```

###  Link to test content
Here's a link to my project:

[Test.zip](https://github.com/google/ExoPlayer/files/3229297/Test.zip)

###  Version of ExoPlayer being used
I'm currently using exoplayer 2.9.1

###  Device(s) and version(s) of Android being used
I expected my app to run on a minimum of Android version 5.0. I also have tested my App on my phone (Samsung S6, Android 7.0) and on a (Samsung s4 mini, Android 6.0)

to summarize the problems:
after 30 - 40 secs a different video playes.
Play Sequence is wrong . (Channel 1) (Channel 1, Channel 2) (Channel 1,2,3) ....

**Comment by tonihei** [comment]
My comment in https://github.com/google/ExoPlayer/issues/5820#issuecomment-492132623 still applies. You create the MediaSources in a loop and add them repeatedly to the same playlist. So this is working as intended.

**Comment by CreeschYT** [comment]
Oh ok now it's fixed thanks !
